### ISSUE TITLE
Merging (concat) with theano and tensorflow gives different tensor

### ISSUE THREAD

**Issue by raun1** [fchollet/keras#6403]
Hello,
So I have been using theano till recently for my backend. During the merge operations when I would use concat the dimensions on one axis would double. Eg is my two tensors were 50x50 and 50x50 the resultant merged tensor would be 50x100,
However now that I am using tensorflow, on combining two tensors 50x50 and 50x50 using concat the output tensor is 50x50. 
Is that normal? Or is there a bug in my code somewhere while using tensorflow.
Thank you!

- [ -] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [- ] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [- ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by alyato** [comment]
@raun1 Is it right using tf? how do you solve it? Thanks.

**Comment by raun1** [comment]
Its just that what I had asked. The way theano and tensorflow implement merge is just different nothing more. Though if you ask me I prefer the tensorflow merge more. So in the case of theano you have to keep in mind about the increase in the dimension and if you need the images to be same size once again you can just do a pooling layer with pool_size=(1,2) if you are merging two layers or (1,3) if you are merging three and so on.

**Comment by alyato** [comment]
@raun1 what does the following mean：

> you can just do a pooling layer with pool_size=(1,2) if you are merging two layers or (1,3) if you are merging three and so on.

if the cov_output is `(none,none,12,12),` the outshape is `(none,none,6,6)` after `MaxPool(2,2)`.
when set the `pool_size=(1,2)`, the outshape is ?

**Comment by raun1** [comment]
The output shape will be (12,6)
